libres/lib/job_queue/rsh_driver.cpp

```diff
@@ -303,29 +310,35 @@ void rsh_driver_add_host(rsh_driver_type *rsh_driver, const char *hostname,
     }
 }
 
+/**
+ * Split RSH hostnames into host and (optionally) max_running
+ *
+ * Eg. If the hostname is "foo:42", return {"foo", 42}
+ * If the hostname is just "foo", return {"foo", 1}
+ *
+ * @param[in] hostname String to split. Eg. "foo:42" or just "foo"
+ * @return Pair of host, max_running.
+ */
+std::pair<std::string, int> rsh_split_host(std::string hostname) {
+    int host_max_running = 1;
+
+    auto pos = hostname.rfind(':');
+    if (pos != hostname.npos) {
+        // found ':'
+        host_max_running = std::stoi(host.substr(pos + 1));
+        hostname = hostname.substr(0, pos);
+    }
+    return {pos, hostname};
+}
+
 /**
    Hostname should be a string as host:max_running, the ":max_running"
    part is optional, and will default to 1.
 */
 void rsh_driver_add_host_from_string(rsh_driver_type *rsh_driver,
```

Comment:
This function is two lines and is used once in `rsh_driver_set_option`. It should be inlined there.